cmd/site-replication.go

```diff
@@ -3443,12 +3446,17 @@ func (c *SiteReplicationSys) EditPeerCluster(ctx context.Context, peer madmin.Pe
 	)
 
 	for _, v := range sites.Sites {
-		if peer.Endpoint == v.Endpoint {
-			return madmin.ReplicateEditStatus{}, errSRInvalidRequest(fmt.Errorf("Endpoint %s entered for deployment id %s already configured in site replication", v.Endpoint, v.DeploymentID))
-		}
 		if peer.DeploymentID == v.DeploymentID {
 			found = true
-			if peer.Endpoint == v.Endpoint {
+			if !peer.SyncState.Empty() {
+				if globalDeploymentID == peer.DeploymentID {
+					return madmin.ReplicateEditStatus{}, errSRInvalidRequest(fmt.Errorf("invalid sync setting for endpoint %s (deployment id %s)", v.Endpoint, v.DeploymentID))
+				}
+				if peer.Endpoint == "" {
```

Comment:
```suggestion
				if peer.Endpoint == "" {
```

when is this expected to happen? can you add a comment

Reply:
added a comment - `mc admin replicate update` can choose to update endpoint or sync/async mode or both.